pkg/oidc/verifier.go

```diff
@@ -130,19 +131,34 @@ func CheckAudience(claims Claims, clientID string) error {
 	return nil
 }
 
+// AZPVerifier specifies the function to be used by the `DefaultVerifier` for validating the azp claim
+type AZPVerifier func(string) error
+
+// DefaultAZPVerifier implements `AZPVerifier` returning an error
+// if none of the provided values matches the azp claim
+func DefaultAZPVerifier(possibleValues ...string) AZPVerifier {
+	return func(azp string) error {
+		if !slices.Contains(possibleValues, azp) {
+			return fmt.Errorf("expected one of: %v, got: %q", possibleValues, azp)
+		}
+		return nil
+	}
+}
+
 // CheckAuthorizedParty checks azp (authorized party) claim requirements.
 //
 // If the ID Token contains multiple audiences, the Client SHOULD verify that an azp Claim is present.
-// If an azp Claim is present, the Client SHOULD verify that its client_id is the Claim Value.
+// If an azp Claim is present, the Client MAY verify that its client_id is the Claim Value.
 // https://openid.net/specs/openid-connect-core-1_0.html#IDTokenValidation
-func CheckAuthorizedParty(claims Claims, clientID string) error {
+func CheckAuthorizedParty(claims Claims, azp AZPVerifier) error {
```

Comment:
It would be great if we can prevent this breaking change in the function signature as well.

I suggest to add a new function `CheckAZPVerifier(claims Claims, azp AZPVerifier) error` the body of the current proposal.

`CheckAuthorizedParty(claims Claims, clientID string) error` can wrap the new function using a `clientID`.

Reply:
Sorry for the delay, I've addressed this and more in b5d7f778a65fbc0f2b4102728ab5b1d05fb9008f